keps/sig-storage/1847-autoremove-statefulset-pvcs/README.md

```diff
@@ -0,0 +1,578 @@
+a provision where the PVC created for a pod(which is part of the StatefulSet) is removed when the Pod 
+is deleted as part of a scale down. Since the subsequent scale up needs to create fresh PVCs, it will 
+be slower than scale ups relying on existing PVCs(from earlier scale ups). 
+
+User would set the `PersistentVolumeClaimReclaimPolicy` as 'RemoveOnScaledown' ensuring PVCs are deleted when corresponding
+Pods are deleted. New Pods created during sclae up followed by a scaledown will wait for freshly created PVCs.
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+This feature applies to PVs which are dynamically provisioned from the volumeClaimTemplate of a StatefulSet. Any PVC and PV provisioned from this mechanism will function with this feature.
+
+### Risks and Mitigations
+
+Currently the PVCs created by statefulset are not deleted automatically. Using the 
+`RemoveOnScaledown` or `RemoveOnStatefulSetDeletion` would delete the PVCs 
+automatically. Since this involves persistent data being deleted, users should take 
+appropriate care using this feature. Having the `Retain` behaviour as default 
+will ensure that the PVCs remain intact by default and only a conscious choice 
+made by user will involve any persistent data being deleted. Also, PVCs associated with the StatefulSet will be more durable than ephemeral volumes would be, as they are only deleted on scaledown or StatefulSet deletion, and not on other pod lifecycle events like being rescheduled to a new node, even with the new retain policies.
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Volume reclaim policy for the StatefulSet created PVCs
+A new field named `PersistentVolumeClaimReclaimPolicy` of the type `StatefulSetPersistentVolumeClaimReclaimPolicy` 
+will be added to the StatefulSet. This field represents the user indication on whether the 
+associated PVCs can be deleted or not. The default policy would be `Retain`.
+
+If `PersistentVolumeClaimReclaimPolicy` is set to `RemoveOnScaledown`, Pod is set as the owner of the PVCs created
```

Comment:
it may be more clear here to say explicitly that an ownerref is created rather than just "pod is set as the owner"

Reply:
I think this could be made more clear by rearranging like his:

A `PersistentVolumeClaimReclaimPolicy` of `RemoveOnScaledown` affects both changing the number of replicas in the StatefulSet as well as StatefulSet deletion. When the controller finds that the number of replicas is being reduced, it identifies the pods that are being deleted as well as any PVCs created from the VolumeClaimTemplate of those pods. The OwnerRef of those PVCs is set to the associated pod. When a Pod is deleted, this causes the associated PVC to be deleted as well. When the `RemoveOnScaledown` policy is set etc (keep what you have written).